src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -1122,41 +1129,55 @@ abstract class MergeIntoSuiteBase
       name: String)(
       mergeOn: String,
       mergeClauses: MergeClause*)(
-      errorStrs: Seq[String]): Unit = {
+      errorStrs: Seq[String],
+      notErrorStrs: Seq[String] = Nil): Unit = {
     test(s"extended syntax analysis errors - $name") {
-      withKeyValueData(source = Seq.empty, target = Seq.empty) { case (sourceName, targetName) =>
-        val e = intercept[AnalysisException] {
+      withKeyValueData(
+        source = Seq.empty,
+        target = Seq.empty,
+        sourceKeyValueNames = ("key", "newValue"),
+        targetKeyValueNames = ("key", "oldValue"),
+      ) { case (sourceName, targetName) =>
+        val errMsg = intercept[AnalysisException] {
           executeMerge(s"$targetName t", s"$sourceName s", mergeOn, mergeClauses: _*)
-        }
-        errorStrs.foreach { s => errorContains(e.getMessage, s) }
+        }.getMessage
+        errorStrs.foreach { s => errorContains(errMsg, s) }
+        notErrorStrs.foreach { s => errorNotContains(errMsg, s) }
       }
     }
   }
 
   testAnalysisErrorsInExtendedMerge("update condition - ambiguous reference")(
     mergeOn = "s.key = t.key",
-    update(condition = "key > 1", set = "key = s.key, value = s.value"))(
+    update(condition = "key > 1", set = "oldValue = newValue"))(
     errorStrs = "reference 'key' is ambiguous" :: Nil)
 
   testAnalysisErrorsInExtendedMerge("update condition - unknown reference")(
     mergeOn = "s.key = t.key",
-    update(condition = "unknownAttrib > 1", set = "key = s.key, value = s.value"))(
-    errorStrs = "UPDATE condition" :: "unknownAttrib" :: Nil)
+    update(condition = "unknownAttrib > 1", set = "oldValue = newValue"))(
+    // Should show unknownAttrib as invalid ref and (key, oldValue, newValue) as valid column names.
+    errorStrs = "UPDATE condition" :: "unknownAttrib" :: "key" :: "oldValue" :: "newValue" :: Nil)
 
   testAnalysisErrorsInExtendedMerge("update condition - aggregation function")(
     mergeOn = "s.key = t.key",
-    update(condition = "max(0) > 0", set = "key = s.key, value = s.value"))(
+    update(condition = "max(0) > 0", set = "oldValue = newValue"))(
     errorStrs = "UPDATE condition" :: "aggregate functions are not supported" :: Nil)
 
   testAnalysisErrorsInExtendedMerge("update condition - subquery")(
     mergeOn = "s.key = t.key",
-    update(condition = "s.value in (select value from t)", set = "key = s.key, value = s.value"))(
+    update(condition = "s.value in (select value from t)", set = "oldValue = newValue"))(
     errorStrs = Nil) // subqueries fail for unresolved reference to `t`
 
+  testAnalysisErrorsInExtendedMerge("delete condition - ambiguous reference")(
```

Comment:
this was missing test. the ambiguous refs in delete condition were not being tested.